### ISSUE TITLE
`map` has inconsistent return type for numbers

### ISSUE THREAD

**Issue by eschnett** [JuliaLang/julia#13113]
Numbers behave in Julia like containers with a single element (see `base/numbers.jl`):
```Julia
julia> map(+, 1)
1
```

However:
```Julia
julia> map(+, 1, 2)
1-element Array{Any,1}:
 3
```
The return type is wrong.

I think this is due to a missing `map` signature in `numbers.jl`.

**Comment by JeffBezanson** [comment]
I think the real bug is the existence of a `map` for any iterators that always returns an `Any` array.

I'm curious --- did a case come up where it was actually useful to have `map` treat numbers like collections?

**Comment by my-little-repository** [comment]
The behavior of map is indeed strange.   

     julia> map(+, 1:4,1)
    1-element Array{Any,1}:
     2
    
    julia> map(+, 1:4,1:1)
    ERROR: DimensionMismatch("dimensions must match")
     in map at abstractarray.jl:1310

**Comment by eschnett** [comment]
@JeffBezanson No, I don't have a real use case. But since it exists, I wanted to support it, and this inconsistency makes it impossible.

**Comment by tkelman** [comment]
closed by #13129?
